src/aegis/src/aegis_server.erl

```diff
@@ -95,7 +101,7 @@ encrypt(#{} = Db, Key, Value) when is_binary(Key), is_binary(Value) ->
                 {error, Reason} ->
                     erlang:error(Reason)
             end;
-        false ->
+        _ ->
```

Comment:
Now that I look at it, we'll be exposing users to a high latency period every 30 minutes that they _don't_ revoke access to their key, which we expect to be the case almost every time.

Like in epep, we had "fresh" "stale" and "expired" to cover this. A fresh item is used directly, as your first clause does. An "expired" one is not used (gets removed from cache if present, expensive external call must happen again). but a "stale" entry _is_ used and also kicks off a check that we still have access (if we do, nothing happens, if we don't, we evict our entry).

Reply:
Unlike epep we can't refresh keys async, because refreshing requires reading a wrapped key from fdb, i.e. it must be done in the same process as the transaction. This means that we are going to have this high latency regardless if we removing key and fetching it on the next read or fetching it proactively once it's "stale".

The only way to avoid this is to add another server process to aegis that'll do refresh of the stale keys in its own separate transactions.